src/net_processing.cpp

```diff
@@ -4751,9 +4914,18 @@ bool PeerManagerImpl::SendMessages(CNode* pto)
                             continue;
                         }
                         if (pto->m_tx_relay->pfilter && !pto->m_tx_relay->pfilter->IsRelevantAndUpdate(*txinfo.tx)) continue;
-                        // Send
+
+                        bool flood_tx = it->second;
+                        if (!recon_state || ((*recon_state).IsChosenForFlooding() && flood_tx)) {
+                            // Always flood to non-reconciliation nodes supporting tx relay.
+                            // For reconciliation nodes, flood if flood_to is set up and transaction is meant for flooding.
+                            vInv.push_back(inv);
+                        } else {
+                            // Storing to populate the reconciliation set.
+                            txs_to_reconcile.push_back(txinfo.tx->GetWitnessHash());
+                        }
```

Comment:
I think you can refactor this code area a bit:
* Substitute `vInv` by a`txs_to_announce` (`std::vector<uint256>`)
* when `vInvTx` is empty or `INVENTORY_BROADCAST_MAX` is reached exit transactions to announce selection
* if peer requires reconciliation, check if its reconciliation set isn't at capacity and drain `txs_to_announce` accordingly
* flood the remainder of `txs_to_announce`, splitting messages accordingly to `MAX_INV_SZ`

That way you conserve announcement pre-checks (feefilter, still-present-in-mempool, ...) in common across tx-announcement strategies while keeping the code path straighter.